core/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -106,14 +110,8 @@ public RecoverySourceHandler(final IndexShard shard, final StartRecoveryRequest
         this.transportService = transportService;
         this.indexName = this.request.shardId().index().name();
         this.shardId = this.request.shardId().id();
-
+        this.chunkSizeInBytes = recoverySettings.getChunkSize().bytesAsInt();
```

Comment:
I would just access the recoverySettings directly - test changes will have effect faster.